api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -599,7 +598,7 @@ public TypeID typeId() {
     }
 
     public String crs() {
-      return crs;
+      return crs != null ? crs : DEFAULT_CRS;
```

Comment:
nit: why don't we handle the null param in the constructor?

Reply:
I remembered that we preferred holding null in member and handle it elsewhere when reviewing the previous PR, see https://github.com/apache/iceberg/pull/12346#discussion_r1995885931